app/src/views/AccountMyFormsDref/DownloadImportTemplateButton/DownloadImportTemplateModal/index.tsx

```diff
@@ -0,0 +1,341 @@
+import {
+    useCallback,
+    useMemo,
+    useState,
+} from 'react';
+import {
+    Button,
+    Modal,
+    RadioInput,
+} from '@ifrc-go/ui';
+import { useTranslation } from '@ifrc-go/ui/hooks';
+import { stringValueSelector } from '@ifrc-go/ui/utils';
+import {
+    isDefined,
+    isNotDefined,
+    listToGroupList,
+    listToMap,
+    mapToList,
+} from '@togglecorp/fujs';
+import xlsx from 'exceljs';
+import FileSaver from 'file-saver';
+
+import useGlobalEnums from '#hooks/domain/useGlobalEnums';
+import {
+    DREF_TYPE_RESPONSE,
+    TypeOfDrefEnum,
+} from '#utils/constants';
+import {
+    DrefSheetName,
+    SHEET_ACTIONS_NEEDS,
+    SHEET_EVENT_DETAIL,
+    SHEET_OPERATION,
+    SHEET_OPERATION_OVERVIEW,
+    SHEET_TIMEFRAMES_AND_CONTACTS,
+} from '#utils/domain/dref';
+import { createImportTemplate } from '#utils/importTemplate';
+import {
+    actionsTabFields,
+    eventDetailTabFields,
+    operationTabFields,
+    overviewTabFields,
+    timeframeAndContactsTabFields,
+} from '#views/DrefApplicationForm/common';
+
+import {
+    addHeadingRow,
+    addInputRow,
+    buildCoverWorksheet,
+    headerRowStyle,
+} from '../utils';
+import useImportTemplateSchema from './useImportTemplateSchema';
+
+import i18n from './i18n.json';
+
+function typeOfDrefKeySelector(option: { key: TypeOfDrefEnum }) {
+    return option.key;
+}
+
+interface Props {
+    onComplete: () => void;
+}
+
+function DownloadImportTemplateModal(props: Props) {
+    const { onComplete } = props;
+
+    const { dref_dref_dref_type } = useGlobalEnums();
+    const strings = useTranslation(i18n);
+
+    const [generationPending, setGenerationPending] = useState(false);
+    const [typeOfDref, setTypeOfDref] = useState<TypeOfDrefEnum>(DREF_TYPE_RESPONSE);
+
+    const { drefFormSchema, optionsMap } = useImportTemplateSchema();
+    const templateActions = createImportTemplate(drefFormSchema, optionsMap);
+
+    const drefTypeLabelMap = useMemo(
+        () => (
+            listToMap(
+                dref_dref_dref_type,
+                (option) => option.key,
+                (option) => option.value,
+            )
+        ),
+        [dref_dref_dref_type],
+    );
+
+    const handleDownloadClick = useCallback(() => {
+        if (isNotDefined(templateActions)) {
+            return;
+        }
+
+        async function generateTemplate() {
```

Comment:
Let's separate this out function out of the component.
Let's also add comments.